src/hotspot/cpu/riscv/vm_version_riscv.cpp

```diff
@@ -216,6 +216,38 @@ void VM_Version::common_initialize() {
     warning("CRC32C intrinsics are not available on this CPU.");
     FLAG_SET_DEFAULT(UseCRC32CIntrinsics, false);
   }
+
+  // UseZvbb (depends on RVV).
+  if (UseZvbb && !UseRVV) {
+    if (!FLAG_IS_DEFAULT(UseZvbb)) {
```

Comment:
So we have two code paths to enable this flag: 1. through the command line; 2. through hwprobe. Here the issue is case 2. I wonder if that could be handled in that code path, that is when we call `UPDATE_DEFAULT` [1].

Then we could only consider case 1 here and simplify the code removing this `FLAG_IS_DEFAULT` check. It's a bit confusing to me as people might think that a true value of `UseZvbb` will mean that `FLAG_IS_DEFAULT` is false. What do you think?

[1] https://github.com/openjdk/jdk/blob/master/src/hotspot/cpu/riscv/vm_version_riscv.hpp#L171

Reply:
I'm not sure exactly what you had in mind, but I pushed what I think you were thinking.